docs/specs/mlnet-database-loader/mlnet-database-loader-specs.md

```diff
@@ -0,0 +1,407 @@
+
+- *Other customers - TBD*
+
+# Who is the Customer
+
+- .NET developers used to work with databases who are getting started with machine learning with ML.NET.
+
+
+# Goals
+
+The business goals are the following, depending on the possible scenarios:
+
+- Ability for developers to load and automatically stream data from relational databases in order to train/evaluate ML.NET models.
+- The code to load from a database should be extremely easy, a single line of code in most cases.
+- Tooling (Model Builder in VS and the CLI) and AutoML API should also support this feature.  
+
+# Solution
+
+The solution is to create an ML.NET database loader classes supporting the above scenarios. 
+
+The main supported features are:
+
+- **Easily load data into IDataView from databases**: The main goal is to load data into IDataView from databases very easily (a single line of code as one of the approaches) and properly (with good performance when streaming data while training).
+
+- **Train ML.NET models with data streaming coming from a database**: As the ultimate goal, the main purpose of loading data into an IDataView is to usually train an ML.NET model with data coming from a database table (or view or SQL sentence).
+
+- **Evaluate ML.NET models with data streaming coming from a database**: As secondary goal is to evaluate the quality/accuracy of an ML.NET model with data coming from a database table (or view or SQL sentence) by comparing it with a set or predictions. This goal should support:
+
+  - Train/test split scenario (one database source as training dataset and one database source as testing-dataset), such as a 80%-20% approach.
+
+  - Train/evaluation/test split scenario (one database source as training dataset, one database source as evaluation-dataset and one database source as test-dataset), such as a 50%-30%-20% approach.
+
+  - Cross-validation scenario. Single database source. Internally it'll be split in multiple folds (such as 5 folds) for multiple trains and tests. This should be transparent from a database connection point of view which only needs one database source.
+
+- **Additional support for AutoML API, CLI and Model Builder:** Loading data from databases should be supported by AutoML API, Model Builder in VS and the ML.NET CLI. 
+
+--------------------------------------
+
+TBD - Include diagram with blocks of AutoML, CLI and Model Buidler using this component/loader
+
+--------------------------------------
+
+
+## Design criteria
+
+- **Supported frameworks in .NET**: The DatabaseLoader component should be supported for the [frameworks supported by ML.NET](https://github.com/dotnet/machinelearning#installation) which include:
```

Comment:
Wouldn't it be easier to only target .NET Standard 2.0?

Reply:
Sure, we'll be using/targeting .NET Standard 2.0 precisely, that was my conversation with Diego Vega. I just  mean that .NET Framework and .NET Core apps should be supported. I'll make that clear in the specs doc.

Reply:
Update in the specs so .NET Standard is mentioned explicitly.